account_payment_order/models/account_move.py

```diff
@@ -52,7 +52,11 @@ def _get_payment_order_communication_direct(self):
         communication = self.payment_reference or self.ref or self.name
         if self.is_invoice():
             if self.is_purchase_document():
-                communication = self.ref or self.payment_reference
+                if self.move_type in self.get_outbound_types(include_receipts=False):
+                    # Vendor bill
+                    communication = self.payment_reference or self.ref
+                else:
+                    communication = self.ref or self.payment_reference
             else:
                 communication = self.payment_reference or self.name
         return communication or ""
```

Comment:
It makes sense to me that the payment reference should have priority in all cases, so there's no need for a special case for the in_refund type.

```python

def _get_payment_order_communication_direct(self):
    """Retrieve the communication string for this direct item."""
    communication = self.payment_reference
    if self.is_invoice():
        if self.is_purchase_document():
            communication = communication or self.ref
        else:
            communication = communication or self.name
    return communication or ""

```

Reply:
I agree and that's what I would prefer, but this breaks another test related to refunds. And since I don't understand why it was done that way I'd like to have opinions of others.

Reply:
I think we should follow @sbejaoui comment.

I don't know why it was done this way :thinking:

Reply:
Wait, wait, please let the customer part. It was done on purpose, when Odoo decided to add in the payment reference a number suffix to the invoice number, so you got an unknown number. For example, you had as invoice number "FV/2024/0001". When posting, Odoo put as payment reference "FV/2024/0001/24".

Fortunately, this is not the case anymore (or at least it seems so), but that field is used a lot by users to fill other information, like some customer references, so changing it will break current flows.